Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -2116,19 +2116,26 @@ private void LogTempDirectoryTooLong(string directory)
         /// </summary>
         private async Task<bool> PrepareOutputs()
         {
-            if (!PrepareDirectoryOutputs())
-            {
-                return false;
-            }
-
-            if (!PrepareInContainerPaths())
-            {
-                return false;
-            }
-
+            using (var preserveOutputWhitelistWrapper = Pools.GetAbsolutePathSet())
             using (var dependenciesWrapper = Pools.GetAbsolutePathSet())
             using (var outputDirectoriesWrapper = Pools.GetAbsolutePathSet())

```

Comment:
What are the odds of this function returning early in a succesfull build? If they are pretty high, maybe we should keep the pool access to the original location.
If low, this is a great change for readability.

Reply:
early returns are all about failed builds, so we do not need to worry about it. 

---
In reply to: [287434293](https://github.com/microsoft/BuildXL/pull/369#discussion_r287434293) [](ancestors = 287434293)